From 032b9d89b00de885a703cabab77e37d9241e15c8 Mon Sep 17 00:00:00 2001
From: Tj <hacker@iam.tj>
Date: Mon, 6 Nov 2023 16:59:52 +0000
Subject: [PATCH] fix: build on v6.6+ kernel

Commit bffcc6882a "genetlink: remove userhdr from struct genl_info"
caused the build to fail since the field no longer exists.

Replace with run-time calculation of the header offset.

Signed-off-by: Tj <linux@iamtj>
---
 src/mod/common/nl/nl_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mod/common/nl/nl_common.c b/src/mod/common/nl/nl_common.c
index 260218b4..86e184fc 100644
--- a/src/mod/common/nl/nl_common.c
+++ b/src/mod/common/nl/nl_common.c
@@ -14,7 +14,7 @@ char *get_iname(struct genl_info *info)
 
 struct joolnlhdr *get_jool_hdr(struct genl_info *info)
 {
-	return info->userhdr;
+	return (struct joolnlhdr *)((u8 *)info->genlhdr + GENL_HDRLEN);
 }
 
 static int validate_magic(struct joolnlhdr *hdr)
